### DISCUSSION TITLE
Clean exit

### DISCUSSION THREAD

**Discussion by israndy** [cbpowell/SenseLink#20] [Q&A]
Still perfecting my TesSense app. I am learning about coroutines and added another task to the app. Since my original task and this new task are running in my code I just added a global that I can change that interrupts both their While() loops. But the SenseLink server task continues running and the app never finishes. Is there a programmatical way to request the server to exit?

**Comment by cbpowell** [comment]
Hey sorry for the slow reply - yeah there should be, but I think it maybe depends on how you're setting it up? For me a Ctrl-C or other keyboard interrupt does end everything, but I'm by no means an expert at asyncio/coroutines.

**Comment by israndy** [comment]
Thanks for getting back to me, I'll continue to look for a way for the app to quit on it's own. I have learned SO MUCH in such a short time, still not considering myself a "programmer" but a hobbiest certainly now.
